tests/python_package_test/test_basic.py

```diff
@@ -84,6 +93,88 @@ def test_basic(tmp_path):
                                    bst.predict, tname)
 
 
+class NumpySequence(lgb.Sequence):
+    def __init__(self, ndarray):
+        self.ndarray = ndarray
+
+    def __getitem__(self, idx):
+        # The simple implementation is just a single "return self.ndarray[idx]"
+        # The following is for demo and testing purpose.
+        if isinstance(idx, int):
+            return self.__get_one_line__(idx)
+        elif isinstance(idx, slice):
+            if not (idx.step is None or idx.step is 1):
+                raise NotImplementedError("No need to implement, caller will not set step by now")
+            return self.ndarray[idx.start: idx.stop]
+        else:
+            raise TypeError("Sequence Index must be an integer/list/slice, got {}".format(type(idx)))
+
+    def __get_one_line__(self, idx):
+        return self.ndarray[idx]
+
+    def __len__(self):
+        return len(self.ndarray)
+
+
+@pytest.mark.parametrize('sample_count', [2, 5])
+@pytest.mark.parametrize('batch_size', [3, 20, None])
+@pytest.mark.parametrize('include_0', [False, True])
+@pytest.mark.parametrize('include_nan', [False, True])
+@pytest.mark.parametrize('num_seq', [1, 3])
+def test_sequence(tmpdir, sample_count, batch_size, include_0, include_nan, num_seq):
+    rm_files(["seq.truth.bin", "seq.seq.bin"])
```

Comment:
It seems that `rm_files` removes `seq.truth.bin` and `seq.seq.bin`, but subsequently `tmpdir/seq.truth.bin` and `tmpdir/seq.seq.bin` will be created. That's a mismatch.
And why do we need to manually remove the files created under `tmpdir` here, which is not done in other testing cases?

Reply:
> why do we need to manually remove the files created under tmpdir here, which is not done in other testing cases?
we don't. tmpdir is unique per parameter.

This is indeed a left-out from previous code where `tmpdir`  was not used.

Will fix.